packit_service/service/api/testing_farm.py

```diff
@@ -134,3 +137,40 @@ def validate_testing_farm_request():
         msg = "Invalid testing farm secret provided"
         logger.warning(msg)
         raise ValidationFailed(msg)
+
+
+@ns.route("/get-results")
+class TestingFarmResultsList(Resource):
+    @ns.expect(pagination_arguments)
+    @ns.response(HTTPStatus.PARTIAL_CONTENT, "Testing Farm Results list follows")
+    def get(self):
+        """ List all Testing Farm  results. """
+
+        result = []
+
+        first, last = indices()
+        # results have nothing other than ref in common, so it doesnt make sense to
+        # merge them like copr builds
+        for tf_result in TFTTestRunModel.get_all(first, last):
+
+            result_dict = {
+                "pipeline_id": tf_result.pipeline_id,
+                "ref": tf_result.commit_sha,
+                "status": tf_result.status,
+                "target": tf_result.target,
+                "web_url": tf_result.web_url,
+                "pr_id": tf_result.get_pr_id(),
+            }
+
+            project = tf_result.get_project()
+            if project:
+                result_dict["repo_namespace"] = project.namespace
+                result_dict["repo_name"] = project.repo_name
+
+            result.append(result_dict)
+
+        resp = make_response(dumps(result), HTTPStatus.PARTIAL_CONTENT)
+        resp.headers["Content-Range"] = f"copr-builds {first + 1}-{last}/{len(result)}"
```

Comment:
From a quick search on this header, this seems to be for transferring big files (so that clients can continue a download), but in this case we talk pagination. So I think this is somewhat misleading HTTP-talk :)

I'll look more into it.

Reply:
Some Git-archeology reveals that using the `Content-Range` header was started in #187, which aimed to fix #185. The later makes some references to some Stackoverflow questions about REST pagination.

Following links from there, it seems to me, that the original intention was to do pagination as described in [this blog post](http://otac0n.com/blog/2012/11/21/range-header-i-choose-you.html), but implementation wasn't really carried out all the way.

Anyways... this doesn't have to be fixed in this PR, but please use `{first+1}-{last}/*` in the answer or get a count of all the test results if that's not to expensive.

Also: the unit here should be `test-results` instead of `copr-builds`.